### ISSUE TITLE
Stopping an animation manually does not report the right position when using the native driver

### ISSUE THREAD

**Issue by Fearnbuster** [facebook/react-native#28252]
Unless I'm mistaken, it would seem that when you stop an animation manually, it does not report the right position when the animation is driven by the native driver.

For example, this will print the value that the animation **started** at, not what it ended at:
```
    const value = new Animated.ValueXY({ x: 100, y: 100 });

    Animated.timing(value, {
      duration: 1000, // 1 second animation
      toValue: { x: 0, y: 0 },
      useNativeDriver: true,
    }).start();

    setTimeout(() => {
      value.stopAnimation((finalValue) => {
        console.log(finalValue); // This will print { x: 100, y: 100 } when it should be { x: 0, y: 0 }
      });
    }, 10000); // Wait 10 seconds
```
However, if I do **not** use the native driver, then the value prints correctly.
The same problem happens no matter how long (I tried 20s) or short (I tried 100ms) I wait.

When not using the native driver, the correct value will print even when I stop the animation before it is finished.

**React Native version:**
    OS: Windows 10 10.0.18363
    CPU: (8) x64 Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz
    Memory: 2.17 GB / 7.91 GB
  Binaries:
    Node: 12.9.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.7.0 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.10.1 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    react: ~16.9.0 => 16.9.0
    react-native: https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz => 0.61.4

**Comment by gdoudeng** [comment]
+1

**Comment by gdoudeng** [comment]
Is there any solution?

**Comment by Fearnbuster** [comment]
I didn't figure out a solution as I ended up not needing the functionality.  

It's late for me here, so I'll see what I can find tomorrow, and I'll let you know.

**Comment by gdoudeng** [comment]
thanks @Fearnbuster

**Comment by Fearnbuster** [comment]
I believe I found a simple solution, although it is very, **very** unintuitive.

Simply add an event listener to the value that is being animated/stopped and see if it fixes the problem.

For example:
```
const value = new Animated.ValueXY({ x: 100, y: 100 }); 

value.addListener(() => {});
```

The event listener doesn't even have to do anything or take any arguments, it simply has to be present.

I noticed that, while trying to find the problem, sometimes when I manually stopped the animation, it would report the right value and sometimes it wouldn't.

Eventually I narrowed it down to the fact that, for testing purposes, sometimes I was adding an event listener and at other times I wasn't.

As to why this fixes the problem (at least for me), I have no idea;  I assume it has something to do with the code after it is compiled.

I hope this fixes your problem.

P.s. I've been working only on my personal phone, which is an Android, so I have no idea if this problem exists on IOS or either of the emulators.  For the same reason, I have no idea if my solution will fix this problem on any of those platforms (if they have the problem, that is).
